### PR TITLE
full functionality of cortical.io Python SDK

### PR DESCRIPTION
**Pull request by BoltzmannBrain**

### CODE REVIEW

**Comment by rhyolight** [comment]
@BoltzmannBrain How about we try to keep our code coverage percentage from decreasing? You should be able to bump the number quite a bit with just one more test.

**Comment by BoltzmannBrain** [comment]
@rhyolight yes yes, thanks for the reminder. Just a lot on my plate but I plan on getting the coverage number into respectable range today and tomorrow... I won't merge this until I add some tests to increase coverage.

**Comment by rhyolight** [comment]
@BoltzmannBrain Thanks!

**Comment by BoltzmannBrain** [comment]
Added some unit tests to cover the changes in this PR, but they are not ready for merge. Individually they pass but when run together I get `Exception: Response 401: Unauthorized Access. Requests must contain a valid api-key.`. Thus the API calls need to be mocked out, which I'll work on with some help from @rhyolight.

**Comment by rhyolight** [comment]
@BoltzmannBrain I started working on an example unit test for `getBitmap()` for you earlier tonight. Should have a PR soon, and might be easier than your approach cause I'm using `httpretty` for the HTTP mocking.

**Comment by BoltzmannBrain** [comment]
@rhyolight it feels good to see that coverage :arrow_double_up: 
Not ready for merge until I push tests that cover the original changes in this PR.

**Comment by rhyolight** [comment]
It's a beautiful thing.

File: tests/unit/mock-api-data/text_androids.json [hunk 1]
```diff
@@ -0,0 +1,1149 @@
+[
+  {
+    "text": "do androids dream of electric sheep",
+    "score": null,
+    "pos_types": null,
+    "fingerprint": {
+      "positions": [
```
Comment (rhyolight) [inline tests/unit/mock-api-data/text_androids.json:7]: This seems wrong. It's this endpoint, right? http://api.cortical.io/Text.htm#!/text/getRepresentationForText_post_0

When I make this request: POST http://api.cortical.io/rest/text

Headers:

```
{"Content-Type": "application/json", "api-key": "mykey", "Accept": "Application/json"}
```

Query params:

```
{"retina_name": "en_synonymous"}
```

Post data:

```
do androids dream of electric sheep
```

I get back a JSON structure like this:

```
[{"positions":[11,34,35,38,40,78,110,125,128,142...]}]
```

Which does not include `text`, `score`, `pos_types`, or `fingerprint`... just `positions`. Also, even the active indices are different.

Comment (BoltzmannBrain) [inline tests/unit/mock-api-data/text_androids.json:7]: Correct, but CorticalClient.getTextBitmap() appends these additional fields manually; this way both getTextBitmap() and getBitmap(), querying different API URLs, return dictionaries with common fields.

**Comment by BoltzmannBrain** [comment]
@rhyolight please review.

File: tests/unit/mock-api-data/text_androids.json (hunk 1 above)
Comment (rhyolight) [inline tests/unit/mock-api-data/text_androids.json:7]: That's fine, but the mock JSON file should be the response as it is returned from the API, not including any local mutations. Asserting that the structure is updated from that raw response by the code under test would be part of the test assertions.

**Comment by rhyolight** [comment]
Even thought the tests pass, it only seems to be because a mock JSON response for `gitTextBitmap()` is incorrect. 

``` python
import cortipy
client = cortipy.CorticalClient(verbosity=2, useCache=False)
client.getTextBitmap("do androids dream of electric sheep")
```

Results in:

```
    fetching 'do androids dream of electric sheep' fingerprint from REST API
Traceback (most recent call last):
  File "test.py", line 14, in <module>
    client.getTextBitmap("do androids dream of electric sheep")
  File "/Users/mtaylor/nta/cortipy/cortipy/cortical_client.py", line 277, in getTextBitmap
    on = len(fpInfo["fingerprint"]["positions"])
KeyError: 'fingerprint'
```

Your unit test expects the API response to contain elements it does not actually contain (per our discussion above), including a `fingerprint` key that is actually added by local code.

**Comment by rhyolight** [comment]
@BoltzmannBrain This is what I'm thinking: https://github.com/BoltzmannBrain/cortipy/pull/1. But I'm confused about why you are adding these extra data to the response.

**Pull request merged**